scout/adapter/mongo/hgnc.py

```diff
@@ -336,28 +336,23 @@ def genes_by_alias(self, build="37", genes=None):
         LOG.debug("Fetching all genes by alias")
         # Collect one entry for each alias symbol that exists
         alias_genes = {}
-        # Loop over all genes
-        if not genes:
-            genes = self.hgnc_collection.find({"build": str(build)})
+
+        if genes is None:
+            genes_query = {"build": str(build)} if build else {}
+            genes = self.hgnc_collection.find(genes_query)
 
         for gene in genes:
-            # Collect the hgnc_id
             hgnc_id = gene["hgnc_id"]
-            # Collect the true symbol given by hgnc
             hgnc_symbol = gene["hgnc_symbol"]
+
             # Loop aver all aliases
```

Comment:
Good with removing the unneeded comments. That one could also go, but not important.